### PR TITLE
MULE-17949: Oracle Data Source Config doesn't allow for additional pooling profile properties.

### PR DESCRIPTION
**Pull request by emariotti3**

### CODE REVIEW

File: src/main/java/org/mule/extension/db/api/config/DbPoolingProfile.java [hunk 1]
```diff
@@ -82,6 +89,11 @@
   @Expression(NOT_SUPPORTED)
   private int maxIdleTime;
 
+  @Parameter
+  @Optional
+  @Placement(tab = ADVANCED_TAB)
+  private Map<String, Object> additionalProperties = emptyMap();
```
Comment (Bardammu) [inline src/main/java/org/mule/extension/db/api/config/DbPoolingProfile.java]: Add a summary of this new parameter:
https://docs.mulesoft.com/mule-sdk/1.1/define-parameters#layout

File: src/main/java/org/mule/extension/db/api/config/DbPoolingProfile.java [hunk 2]
```diff
@@ -6,14 +6,21 @@
  */
 package org.mule.extension.db.api.config;
 
+import static java.util.Collections.emptyList;
```
Comment (Bardammu) [inline src/main/java/org/mule/extension/db/api/config/DbPoolingProfile.java]: Is this import used?

**Comment by Bardammu** [comment]
Try this in Studio and take a picture to upload add it in the PR.
In this way, we can be sure that it looks right on Studio.

**Comment by emariotti3** [comment]
Screen Shot 2020-01-15 at 2 12 10 PM
Screen Shot 2020-01-15 at 2 12 16 PM

**Comment by emariotti3** [comment]
Screen Shot 2020-01-15 at 3 04 24 PM

File: src/main/java/org/mule/extension/db/internal/domain/connection/DataSourceFactory.java [hunk 3]
```diff
@@ -112,7 +114,14 @@ protected DataSource createPooledStandardDataSource(DataSource dataSource, DbPoo
     config.put("maxIdleTime", poolingProfile.getMaxIdleTime());
     config.put("testConnectionOnCheckout", "true");
     config.put("maxStatementsPerConnection", poolingProfile.getPreparedStatementCacheSize());
-
+    poolingProfile.getAdditionalProperties().entrySet().forEach((param) -> {
+      if (!config.containsKey(param.getKey())) {
+        config.put(param.getKey(), param.getValue());
+      } else {
+        LOGGER.warn(format("Attempted to override property {0} using additional-properties. Proceeding to use {0} = {1}.",
```
Comment (Bardammu) [inline src/main/java/org/mule/extension/db/internal/domain/connection/DataSourceFactory.java]: I think `format` is not necessary here. If so, remove it from the import too.

**Pull request merged**